models/dc_generator.cpp

```diff
@@ -111,12 +165,30 @@ nest::dc_generator::calibrate()
 void
 nest::dc_generator::update( Time const& origin, const long from, const long to )
 {
+  assert(
+    to >= 0 && ( delay ) from < kernel().connection_manager.get_min_delay() );
+  assert( from < to );
+
   long start = origin.get_steps();
 
   CurrentEvent ce;
   ce.set_current( P_.amp_ );
 
   for ( long offs = from; offs < to; ++offs )
+  {
+    B_.logger_.record_data( origin.get_steps() + offs );
```

Comment:
In `sinusoidal_poisson_generator`, the recorded variable `S_.rate_` is updated in lines 256-265, so once we get to 268, it has its value at the end of the time step; but the logging in that model ignores the possibility that the rate is zero while the device is inactive. Could you correct that there (and in the corresponding gamma generator?).

The line needs to move down to after the `if` so that it records the correct value (0 or amp) for the end of the time step.